driver-core/src/main/com/mongodb/internal/connection/Authenticator.java

```diff
@@ -103,13 +103,13 @@ abstract void authenticateAsync(InternalConnection connection, ConnectionDescrip
             OperationContext operationContext, SingleResultCallback<Void> callback);
 
     public void reauthenticate(final InternalConnection connection, final OperationContext operationContext) {
-        authenticate(connection, connection.getDescription(), operationContext);
+        authenticate(connection, connection.getDescription(), operationContext.withSessionContext(NoOpSessionContext.INSTANCE));
     }
 
     public void reauthenticateAsync(final InternalConnection connection, final OperationContext operationContext,
                                     final SingleResultCallback<Void> callback) {
         beginAsync().thenRun((c) -> {
-            authenticateAsync(connection, connection.getDescription(), operationContext, c);
+            authenticateAsync(connection, connection.getDescription(), operationContext.withSessionContext(NoOpSessionContext.INSTANCE), c);
         }).finish(callback);
     }
```

Comment:
The `OperationContext` for the first authentication when we establish a connection comes from [`MongoClusterImpl.OperationExecutorImpl.getOperationContext`](https://github.com/mongodb/mongo-java-driver/blob/7e94bc28f7264beb0a4fb62ae32d08122869838c/driver-sync/src/main/com/mongodb/client/internal/MongoClusterImpl.java#L492-L498), and does not have a `SessionContext`.

Reply:
Started using `ReadConcernAwareNoOpSessionContext` in df84c84925d536154cc9650d613c66f8bc1e690d just in case.